File: solutions/go/secret-handshake/1/secret_handshake.go
package secret

func Handshake(code uint) []string {
	result := []string{}
	if code&0b1 != 0 {
		result = append(result, "wink")
	}
	if code&0b10 != 0 {
		result = append(result, "double blink")
	}
	if code&0b100 != 0 {
		result = append(result, "close your eyes")
	}
	if code&0b1000 != 0 {
		result = append(result, "jump")
	}
	if code&0b10000 != 0 {
		tmp01 := []string{}
		for _, i := range result {
			tmp01 = append([]string{i}, tmp01...)
		}
		result = tmp01
	}
	return result
}

File: solutions/cpp/secret-handshake/3/secret_handshake.cpp
#include "secret_handshake.h"

namespace secret_handshake {
std::vector<std::string> commands(int test) {
  std::array<std::string, 4> const respond = {"wink", "double blink",
                                              "close your eyes", "jump"};
  std::vector<std::string> result{};
  for (auto i = 0; i < 4; i++) {
    if (test & (1 << i)) {
      result.push_back(respond[i]);
    }
  }
  if (test & 0x10) {
    std::reverse(result.begin(), result.end());
  }

  return result;
}
}  // namespace secret_handshake
